File: output/ChatGPT/codenet/C++/Python/s970719454.py
while True:
    n, m = map(int, input().split())
    if n == 0:
        break

    here = [0] * 100001
    dp1 = [0] * 100001
    dp2 = [0] * 100001

    dpflg = False

    for i in range(m):
        a = int(input())
        if a == 0:
            dpflg = True
        here[a] = 1

    maxi1 = 0
    maxi2 = 0

    for i in range(1, n + 1):
        if here[i] == 1:
            dp1[i] = dp1[i - 1] + 1
        maxi1 = max(dp1[i], maxi1)

    for i in range(1, n + 1):
        if here[i] == 1:
            dp2[i] = dp2[i - 1] + 1
        else:
            dp2[i] = dp1[i - 1] + 1
        maxi2 = max(dp2[i], maxi2)

    if not dpflg:
        print(maxi1)
    else:
        print(maxi2)

File: output/StarCoder/codenet/C++/Java/s970719454.java
import java.util.*;

public class s970719454 {

	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		while(true){
			int n = sc.nextInt();
			int m = sc.nextInt();
			if(n==0)break;
			
			boolean dpflg = false;
			
			int[] here = new int[100001];
			int[] dp1 = new int[100001];
			int[] dp2 = new int[100001];
			
			rep(i,100001)here[i]=dp1[i]=dp2[i]=0;
			
			rep(i,m){
				int a = sc.nextInt();
				if(a==0) dpflg=true;
				
				here[a]=1;
			}
			
			int maxi1 = 0;
			int maxi2 = 0;
			
			rep(i,1,n+1){
				if(here[i]==1)dp1[i] = dp1[i-1]+1;
				maxi1 = max(dp1[i], maxi1);
			}
			
			rep(i,1,n+1){
				if(here[i]==1)dp2[i] = dp2[i-1]+1;
				else dp2[i] = dp1[i-1]+1;
				maxi2 = max(dp2[i], maxi2);
			}
			
			if(!dpflg){
				System.out.println(maxi1);
			}else{
				System.out.println(maxi2);
			}
		}
	}

}